src/Core/Domain/Cart/Query/GetCartForOrderCreation.php

```diff
@@ -32,13 +32,23 @@
 /**
  * Query for getting cart information
  */
-class GetCartInformation
+class GetCartForOrderCreation
 {
     /**
      * @var CartId
      */
     private $cartId;
 
+    /**
+     * @var bool
+     */
+    private $separateGiftProducts = false;
+
+    /**
+     * @var bool
+     */
+    private $separateGiftCartRules = false;
```

Comment:
I think you're mixing things up, separating gift products and separating gift cart rules is the same thing, it's the same feature so no need for two options for the same split
The two options we mentioned were:
- separate gift
- hide/modify free shipping

So at least `separateGiftCartRules` is redundant and should be removed, about free shipping as I said in an earlier comment we could handle it with `separateGift` via a single option but then it should be renamed